examples/polygon.ml

```diff
@@ -0,0 +1,28 @@
+open Joy.Shape
+
+let rec range a b = if a > b then [] else a :: range (a +. 1.) b
+let num_points = 16.
+let size = 100
+
+let idx_to_point i =
+  let x = cos (2. *. Float.pi *. i /. num_points) in
+  let y = sin (2. *. Float.pi *. i /. num_points) in
+  {
+    x = int_of_float (x *. float_of_int size);
+    y = int_of_float (y *. float_of_int size);
+  }
+
+let () =
+  init ();
+  let points = List.map idx_to_point (range 0. num_points) in
+  let _rect =
+    [
+      { x = 0; y = 0 };
+      { x = 0; y = size };
+      { x = size; y = size };
+      { x = size; y = 0 };
+    ]
+  in
+  let poly = polygon points in
+  render_shape poly;
+  close ()
```

Comment:
let's go with a simple example for our polygon shape.

i feel the base example should make it obvious how to use something without needing to read the rest of the doc -- e.g. what the types of the arguments should be, which in this case, we need to read multiple functions to understand.

can you make a simple polygon with hardcoded values for this? as for what to create is up to you. 

feel free to add this existing example under another name (e.g. whatever it ends up looking like).